src/interface/DistrictModel.cpp

```diff
@@ -0,0 +1,257 @@
+#include "DistrictModel.h"
+
+#include <QDebug>
+
+class DistrictModelPrivate {
+public:
+    District::List districts;
+	bool codeVisible;
+
+public:
+	DistrictModelPrivate()
+		: codeVisible( false )
+	{
+	}
+};
+
+DistrictModel::DistrictModel( QObject* parent )
+    : QAbstractTableModel( parent ),
+        d( new DistrictModelPrivate )
+{
+}
+
+DistrictModel::~DistrictModel()
+{
+    delete d;
+}
+
+int DistrictModel::columnCount( const QModelIndex& parent ) const
+{
+    return parent == QModelIndex() ? DistrictModel::LastRole -DistrictModel::FirstRole : 0;
+}
+
+int DistrictModel::rowCount( const QModelIndex& parent ) const
+{
+    return parent == QModelIndex() ? d->districts.count() : 0;
+}
+
+QVariant DistrictModel::data( const QModelIndex& index, int role ) const
+{
+    if ( index.isValid() ) {
+        const District& district = this->district( index );
+        
+        switch ( role ) {
+            case Qt::DisplayRole:
+            case Qt::EditRole: { 
+                switch ( index.column()  ) {
+                    case 0:
+                        return d->codeVisible ? QString( "%1 (%2)" ).arg( district.label() ).arg( district.code() ) : district.label();
+                    case 1:
+                        return district.value();
+                    case 2:
+                        return district.code();
+                }
+                
+                break;
+            }
+            
+            case DistrictModel::LabelRole: {
+                return district.label();
+            }
+            
+            case DistrictModel::ValueRole: {
+                return district.value();
+            }
+            
+            case DistrictModel::CodeRole: {
+                return district.code();
+            }
+        }
+    }
+    
+    return QVariant();
+}
+
+QVariant DistrictModel::headerData( int section, Qt::Orientation orientation, int role ) const
+{
+    if ( orientation == Qt::Horizontal && role == Qt::DisplayRole && section >= 0 && section < columnCount() ) {
+        switch ( section +DistrictModel::FirstRole ) {
+            case DistrictModel::LabelRole:
+                return tr( "Label" );
+            case DistrictModel::ValueRole:
+                return tr( "Value" );
+            case DistrictModel::CodeRole:
+                return tr( "Code" );
+            default:
+                break;
+        }
+    }
+    
+    return QAbstractTableModel::headerData( section, orientation, role );
+}
+
+void DistrictModel::sort( int column, Qt::SortOrder order )
+{
+	Q_UNUSED( column );
+	
+	if ( rowCount() == 0 ) {
+		return;
+	}
+	
+	emit layoutAboutToBeChanged();
+	
+	const QModelIndexList oldIndexes = persistentIndexList();
+	QModelIndexList newIndexes;
+	
+	QMap<int, District> oldMapping;
+	QMap<District, int> newMapping;
+	
+	for ( int i = 0; i < d->districts.count(); i++ ) {
+		oldMapping[ i ] = d->districts[ i ];
+	}
+	
+	switch ( order ) {
```

Comment:
indentation, globally, u can try QtCreator Ctrl+I shortcut on the complete document selection, that will re-indent all correctly using the project settings.